From 7329c58a8d07f64cea150d703b4b0b99a60d6e55 Mon Sep 17 00:00:00 2001
From: Kohei Yoshida <kyoshida@novell.com>
Date: Tue, 23 Feb 2010 07:57:36 -0500
Subject: [PATCH] Fixed failed date detection that forced input date to 0.

* patches/dev300/svtools-uninitialized-member-fix.diff: the old format
  type was incorrectly reset which caused input date to become '0' in
  the Hungarian locale if the number format was pre-set. (n#581634)
---
 .../dev300/svtools-uninitialized-member-fix.diff   |   21 +++++++++++--------
 1 files changed, 12 insertions(+), 9 deletions(-)

diff --git a/patches/dev300/svtools-uninitialized-member-fix.diff b/patches/dev300/svtools-uninitialized-member-fix.diff
index 2af7878..e538739 100644
--- a/patches/dev300/svtools-uninitialized-member-fix.diff
+++ b/patches/dev300/svtools-uninitialized-member-fix.diff
@@ -1,12 +1,15 @@
 diff --git svtools/source/numbers/zforfind.cxx svtools/source/numbers/zforfind.cxx
-index 9fba39e..872ada2 100644
+index 9fba39e..fbf1dc1 100644
 --- svtools/source/numbers/zforfind.cxx
 +++ svtools/source/numbers/zforfind.cxx
-@@ -138,6 +138,7 @@ void ImpSvNumberInputScan::Reset()
-     nAnzNums     = 0;
-     nThousand    = 0;
-     eScannedType = NUMBERFORMAT_UNDEFINED;
-+    eSetType     = NUMBERFORMAT_UNDEFINED;
-     nAmPm        = 0;
-     nPosThousandString = 0;
-     nLogical     = 0;
+@@ -87,7 +87,9 @@ ImpSvNumberInputScan::ImpSvNumberInputScan( SvNumberFormatter* pFormatterP )
+         pUpperMonthText( NULL ),
+         pUpperAbbrevMonthText( NULL ),
+         pUpperDayText( NULL ),
+-        pUpperAbbrevDayText( NULL )
++        pUpperAbbrevDayText( NULL ),
++        eScannedType( NUMBERFORMAT_UNDEFINED ),
++        eSetType( NUMBERFORMAT_UNDEFINED )
+ {
+     pFormatter = pFormatterP;
+     pNullDate = new Date(30,12,1899);
-- 
1.7.0

